tests/src/JIT/Directed/arglist/vararg.cs

```diff
@@ -0,0 +1,1195 @@
+    /// then return the value.
+    /// 
+    /// Do a quick check on the value returned, and return whether they equal.
+    /// 
+    /// </summary>
+    /// <param name="expectedValues"></param>
+    /// <returns>bool</returns>
+    static bool TestPassingLongs(long[] expectedValues)
+    {
+        long expectedSum = test_passing_longs(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+        long sum = 0;
+        for (int i = 0; i < expectedValues.Length; ++i)
+        {
+            sum += expectedValues[i];
+        }
+
+        return sum == expectedSum;
+    }
+
+    /// <summary>
+    /// Given an input set create an arglist to pass to a vararg C function.
+    /// 
+    /// The C function will simply loop over the arguments, compute the sum
+    /// then return the value.
+    /// 
+    /// Do a quick check on the value returned, and return whether they equal.
+    /// 
+    /// </summary>
+    /// <param name="expectedValues"></param>
+    /// <returns>bool</returns>
+    static bool TestPassingFloats(float[] expectedValues)
+    {
+        float expectedSum = test_passing_floats(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+        float sum = 0;
+        for (int i = 0; i < expectedValues.Length; ++i)
+        {
+            sum += expectedValues[i];
+        }
+
+        return sum == expectedSum;
+    }
+
+    /// <summary>
+    /// Given an input set create an arglist to pass to a vararg C function.
+    /// 
+    /// The C function will simply loop over the arguments, compute the sum
+    /// then return the value.
+    /// 
+    /// Do a quick check on the value returned, and return whether they equal.
+    /// 
+    /// </summary>
+    /// <param name="expectedValues"></param>
+    /// <returns>bool</returns>
+    static bool TestPassingDoubles(double[] expectedValues)
+    {
+        double expectedSum = test_passing_doubles(expectedValues.Length, __arglist(expectedValues[0], expectedValues[1], expectedValues[2], expectedValues[3]));
+
+        double sum = 0;
+        for (int i = 0; i < expectedValues.Length; ++i)
+        {
+            sum += expectedValues[i];
+        }
+
+        return sum == expectedSum;
+    }
+
+    /// <summary>
+    /// Given an input set create an arglist to pass to a vararg C function.
+    /// 
+    /// The C function will simply loop over the arguments, compute the sum
+    /// then return the value.
+    /// 
+    /// Do a quick check on the value returned, and return whether they equal.
+    /// 
+    /// </summary>
+    /// <param name="expectedValues"></param>
+    /// <returns>bool</returns>
+    static bool TestPassingEmptyInts(int[] expectedValues)
+    {
+        int expectedSum = test_passing_ints(expectedValues.Length, __arglist());
+
+        int sum = 0;
+        for (int i = 0; i < expectedValues.Length; ++i)
```

Comment:
I know this is just copy-paste template code, but it's weird to loop even though you expect `expectedValues.Length == 0`